### PR TITLE
Create the powershell proxy during execute in order to better handle exceptions.

### LINKED ISSUE #11: Cannot find the file specified, Kernel died
**In the console** The system cannot find the file specified
**In the browser** The kernel appears to have died. It will restart automatically.

**Content of ~/AppData/Roaming/jupyter/kernels/powershell/kernel.json**

`{ "argv": [ "C:\\archive2\\Vinayak\\rnd\\platform\\anaconda\\python.exe", "-m", "powershell_kernel", "-f", "{connection_file}" ], "display_name": "PowerShell", "env": {"powershell_command": "pwsh"}, "language": "powershell" }`

**Full traceback in console.**

> 
> Traceback (most recent call last):
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\runpy.py", line 193, in _run_module_as_main
> "main", mod_spec)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\runpy.py", line 85, in run_code
> exec(code, run_globals)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\powershell_kernel_main.py", line 3, in 
> IPKernelApp.launch_instance(kernel_class=PowerShellKernel)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\traitlets\config\application.py", line 657, in launch_instance
> app.initialize(argv)
> File "", line 2, in initialize
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\traitlets\config\application.py", line 87, in catch_config_error
> return method(app, *args, **kwargs)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\ipykernel\kernelapp.py", line 454, in initialize
> self.init_kernel()
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\ipykernel\kernelapp.py", line 365, in init_kernel
> user_ns=self.user_ns,
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\traitlets\config\configurable.py", line 412, in instance
> inst = cls(*args, **kwargs)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\powershell_kernel\kernel.py", line 50, in init
> repl = subprocess_repl.SubprocessRepl([powershell_command, '-noprofile', '-File', '-'])
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\site-packages\powershell_kernel\subprocess_repl.py", line 29, in init
> stderr=subprocess.STDOUT, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\subprocess.py", line 707, in init
> restore_signals, start_new_session)
> File "C:\archive2\Vinayak\rnd\platform\anaconda\lib\subprocess.py", line 990, in _execute_child
> startupinfo)
> FileNotFoundError: [WinError 2] The system cannot find the file specified
> [I 14:30:56.125 NotebookApp] KernelRestarter: restarting kernel (1/5)
> WARNING:root:kernel 334f9cf8-1836-4623-bf40-a25fb4ecc6a2 restarted

### PR DESCRIPTION
**Pull request by corivera**
Fixes https://github.com/vors/jupyter-powershell/issues/11

Currently the kernel hangs if powershell isn't installed, since an exception is being thrown during init. My solution to this is to move the proxy creation to do_execute so that we can return a well-formatted message.

A nice side-effect of putting this check in do_execute is that if powershell isn't found, then the user can install it and run the cell again successfully without having to restart the kernel/notebook.

### CODE REVIEW

**Comment by vors** [comment]
@chlafreniere removed myself as a reviewer, please feel free to drive this as you see appropriate!

**Comment by chlafreniere** [comment]
Will do. Thanks!

File: powershell_kernel/kernel.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import re
 import signal
 import urllib
+import sys
+import traceback
```
Comment (kevcunnane) [inline powershell_kernel/kernel.py:11]: Just verifying - does this import work in all supported python versions?

File: powershell_kernel/kernel.py [hunk 2]
```diff
@@ -35,9 +37,9 @@ def banner(self):
                      'mimetype': 'text/x-sh',
                      'file_extension': '.ps1'}
 
-    def __init__(self, **kwargs):
```
Comment (kevcunnane) [inline powershell_kernel/kernel.py:38]: Can you explain to me why __init__ isn't needed? I thought the constructor was essentially required (it's been a while for me and Python, so might be wrong).

File: powershell_kernel/kernel.py [hunk 3]
```diff
@@ -51,15 +53,30 @@ def __init__(self, **kwargs):
 
     def do_execute(self, code, silent, store_history=True,
                    user_expressions=None, allow_stdin=False):
-        if not code.strip():
+        try:
+            if not code.strip():
+                return {'status': 'ok', 'execution_count': self.execution_count,
```
Comment (kevcunnane) [inline powershell_kernel/kernel.py:58]: So if cell is empty it'll succeed even though Powershell isn't installed? Just verifying that's the expected behavior (is that consistent with some other kernel like SparkMagic, for example)?

**Review by kevcunnane — APPROVED** [review]
This looks good to me. A few clarifying questions, but this seems like a nice implementation overall.

File: powershell_kernel/kernel.py (hunk 1 above)
Comment (corivera) [reply powershell_kernel/kernel.py:11]: Yes, traceback is supported back to at least 2.7.

File: powershell_kernel/kernel.py (hunk 2 above)
Comment (corivera) [reply powershell_kernel/kernel.py:38]: Without the proxy creation methods the only thing here would be calling super.__init__. There's already an init defined in the parent KernelBase class.

**Review by chlafreniere — APPROVED** [review]

File: powershell_kernel/kernel.py (hunk 3 above)
Comment (corivera) [reply powershell_kernel/kernel.py:58]: Sparkmagic also does nothing if the cell is empty.

**Pull request merged**